utilities/pipelines/staticValidation/compliance/helper/helper.psm1

```diff
@@ -341,3 +341,83 @@ function Get-WorkflowEnvVariablesAsObject {
 
     return $envVars
 }
+
+<#
+Get a list of all versioned parents of a module
+
+.DESCRIPTION
+Get a list of all versioned parents of a module. The function will recursively search the parent directories of the given path until it finds a directory containing a version.json file or reaches the root path.
+The function will return a list of all the directories in the parent hierarchy that contain a version.json file, including the given path.
+The function will return the list in the order from the root path to the given path.
+
+.PARAMETER Path
+Mandatory. The path of the module to search for versioned parents.
+
+.PARAMETER RootPath
+Optional. The root path to stop the search at. The function will not search above this path.
+
+.PARAMETER Filter
+Optional. Only include module folders in the list (i.e., modules with a `main.json` file), or versioned modules folders (i.e., modules with `version.json` files). The default is to include all folders).
+
+.EXAMPLE
+Get-ParentFolderPathList -Path 'C:/bicep-registry-modules/avm/res/storage/storage-account/blob-service/container/immutability-policy'
+
+Get all parent folders of the 'immutability-policy' folder up to 'res'. Returns
+
+- <repoPath>\avm\res\storage
+- <repoPath>\avm\res\storage\storage-account
+- <repoPath>\avm\res\storage\storage-account\blob-service
+- <repoPath>\avm\res\storage\storage-account\blob-service\container
+
+.EXAMPLE
+Get-ParentFolderPathList -Path '.\avm\res\storage\storage-account\blob-service\container\immutability-policy' -RootPath 'C:/bicep-registry-modules/avm/res' -Filter 'OnlyVersionedModules'
+
+Get all versioned parent module folders of the 'immutability-policy' folder up to 'res'. Returns, e.g.,
+- <repoPath>\avm\res\storage\storage-account
+#>
+
+.EXAMPLE
+Get-VersionedParentPathList -Path 'C:/bicep-registry-modules/avm/res/storage/storage-account/blob-service/container' -RootPath 'C:/bicep-registry-modules/avm/res'
+
+...
+#>
+function Get-ParentFolderPathList {
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string] $Path,
+
+        [Parameter(Mandatory = $false)]
+        [string] $RootPath = $repoRootPath,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('OnlyModules', 'OnlyVersionedModules', 'All')]
+        [string] $Filter = 'All'
+
+    )
+
+    $Item = Get-Item -Path $Path
```

Comment:
Is redundant as you're always only using the `FullName` property which equals the `$Path` parameter. In other words, it makes the script 'slower' without providing an added capability

Reply:
It's used for $Item.Parent.FullName below. Would you prefer to explicitly get the parent there instead? I don't have a preference

Reply:
I think we may be talking past one another.

The parameter `Path` has the same value as `$item.FullName`, that is, the path to the folder. 

So my point would be that instead of using `$item` you should be able to replace all locations of `$item.FullName` with `$Path` - unless I'm missing something

Reply:
As discussed offline $Item is used both to get $Item.FullPath (same as Path) and $Item.Parent.FullPath (cam be retrieved via Split-Path as well). I'm happy to use whatever less costly between get-Item and Split-Path